lib/perl/Genome/SoftwareResult.pm

```diff
@@ -820,16 +820,25 @@ sub _lock {
         max_try => 2,
     );
     if (!$lock && $wait) {
-        $class->debug_message("This data set is still being processed by its creator.  Waiting for existing data lock...");
-        $lock = Genome::Sys::LockProxy->new(
-            resource => $resource_lock_name,
-            scope => 'site',
-        )->lock(
-            wait_announce_interval => 600,
-        );
-        unless ($lock) {
-            $class->error_message("Failed to get existing data lock!");
-            die($class->error_message);
+        if (Genome::Config::get("software_result_async_locking")) {
+            my $exit_code = Genome::Config::get(
+                "software_result_async_locking_exit_code");
```

Comment:
This can be simplified to
```
if (my $exit_code = Genome::Config::get('software_result_async_locking')) {
```

The way this is structured, it means if you want it to actually exit early with a 0 exit code, you'd have to specify this value as "0 but true"